README.md
# SSL Grins #

This is the development repository for the SSL Grins plugin.

## Documentation

* [WP Readme](readme.txt)
* [Changelog](changelog.txt)

## Links

* [Plugin Site](http://halfelf.org/plugins/wp-grins-ssl/)
* [Basic Support](http://wordpress.org/support/plugin/wp-grins-ssl)
* [Donate](https://store.halfelf.org/donate/)

changelog.txt
= 5.2.3 =
* 2014 Oct 4, by Ipstenu
* Adjusting check for [New Smilies](https://github.com/avryl/new-smileys)' new smilies to cover for people who leave the folder named 'new-smilies-master' - THANKS GITHUB!

=5.2.2 =
* 2014 May 15, by Ipstenu
* Better checks, less jank, proper hiding
* New smilies hides "uneasy" as well due to JS being a turd

= 5.2.1 =
* 2014 Apr 22, by Ipstenu
* Adjusting check for [New Smilies](https://github.com/avryl/new-smileys)' new smilies.
* Cleaned janky hide, since it's only for New Smilies

= 5.2 =
* 2014 Mar 25, by Ipstenu
* Fixed comment default on to actually be, oh, on.
* Added in check for 'new' smilies (If you're using [New Smilies](https://github.com/avryl/new-smileys) or, in the future, Jetpack or core, hard to say )
* Hides specific new smilies ... kind of. It's janky.
* Dropping support for old WP.

= 5.1.1 =
* 2013 Dec 9, by Ipstenu
* Tweak to bbPress call because JJJ hates me (not really)
* Force prevent any code from running on login pages.
* Proper Internationalization

= 5.1 =
* 2013 May 14, by Ipstenu
* Small conflict with one of my plugins
* Major conflict with font-emoticons

= 5.0 =
* 2013 May 13, by Ipstenu
* Internationalization
* Proper singletons and checking
* Better handling of options
* Link to settings on plugins page

= 4.4 =
* 2013 January 20, by Ipstenu
* Allowing for filters so you can change the images.

= 4.3 =
* 2012 June 18, by Ipstenu
* bbPress 2.1's Fancy Editor borks this.

= 4.2.1 =
* 2012 June 02, by Ipstenu
* Typos. A-freaking-gain

= 4.2 =
* 2012 June 02, by Ipstenu
* bbPress refining (testing with 2.1 as well since that's a thing)

= 4.1 =
* 2012 May 07, by Ipstenu
* bbPress is an option.

= 4.0 =
* 27 April 2012 by Ipstenu
* Major bump. Why? Oh this little thing I like to call bbPress. Fully supported.

= 3.1 =
* 17 April 2012 by Ipstenu
* Trying to fix why it decides, magically, not to show up. Sometimes. Should work. Works on 3.4...

= 3.0 =
* 22 February 2012 by Ipstenu
* Reverted back to Alex's use of Prototype because something went funky cold medina with ajax.

= 2.2 =
* 1 October 2011 by Ipstenu
* Minification for faster everythinging.

= 2.1 =
* 22 September 2011 by Ipstenu
* Removing PHP 4 support (WP doesn't).
* Cleaning out unused aspects from removal of admin panel.

= 2.0 =
* 21 September 2011 by Ipstenu
* Fixed issue with SSL adminstration causing the icons not to show up.
* Removed admin panel. It's 2011. If your theme doesn't have wp_head, you have other issues.
* Consolidated stylesheets (dropping support for IE 6).
* Renamed files and pulled into one directory.
* Compressed files.

= 1.1 =
* 04 November 2009 by Ronalfy
* Fixed bug where grins would show up in the comments panel.
* Re-did the JavaScript to it's completely separate and only runs when necessary.
* Added admin panel option to manual insert smilies on a page or a post.

= 1.0 =
* 25 October 2009 by Ronalfy
* First release.  Yay!

wp-grins.js
jQuery(document).ready(function() {
var $j = jQuery;
$j.wpgrins = {
	grin: function(tag) {
		var myField;
		var value = '';
		if ($j('#content:input').length > 0)  {
			myField = $j('#content:input');
			value = $j("#content:input").attr("value");
			if ($j('#postdivrich') && typeof tinyMCE != 'undefined' && (!$j('#edButtons') || $j('#quicktags')[0].style.display == 'none')) {
				tinyMCE.execCommand('mceInsertContent', false, ' ' + tag + ' ');
				tinyMCE.execCommand('mceRepaint');
				return;
			}
		}
		else if ($j('#comment:input').length > 0) {
			myField = $j('#comment:input');
			value = $j("#comment:input").attr("value");
		}
		else if ($j('#posttext:input').length > 0) {
			myField = $j('#posttext:input');
			value = $j("#posttext:input").attr("value");
		}
		else if ($j('#bbp_topic_content:input').length > 0) {
			myField = $j('#bbp_topic_content:input');
			value = $j("#bbp_topic_content:input").attr("value");
		}
		else if ($j('#bbp_reply_content:input').length > 0) {
			myField = $j('#bbp_reply_content:input');
			value = $j("#bbp_reply_content:input").attr("value");
		}
		else {
			return false;
		}
		if (value == undefined) { value = ''; }
		if (document.selection) {
			myField.focus();
			sel = document.selection.createRange();
			sel.text = ' ' + tag + ' ';
			myField.focus();
		}
		else if (myField[0].selectionStart || myField[0].selectionStart == '0') {
			var startPos = myField[0].selectionStart;
			var endPos = myField[0].selectionEnd;
			var cursorPos = endPos;
			myField.attr("value", value.substring(0, startPos)
							+ ' ' + tag + ' '
							+ value.substring(endPos, value.length));
			cursorPos += tag.length + 2;
			myField.focus();
			myField[0].selectionStart = cursorPos;
			myField[0].selectionEnd = cursorPos;
		}
		else {
			myField.attr("value", value + ' ' + tag + ' ');
			myField.focus();
		}
	},
	init: function() {
		var s = {};
		s.response = 'ajax-response';
		s.type = "POST";
		s.data = $j.extend(s.data, {action: 'grins'});
		s.global = false;
		s.url = wpgrinsssl.Ajax_Url;
		s.timeout = 30000;
		s.success = function(r) {
			var grinsDiv = '<div id="wp_grins">'+r+'</div>';
			if ($j('#postdiv').length > 0) {
				var type = 'after';
				var node = $j('#postdiv');
			}	else if ($j('#postdivrich').length > 0) {
				var type = 'after';
				var node = $j('#postdivrich');
			} else if ($j('#comment').length > 0) {
				var type = 'before';
				var node = $j('#comment');
			} else if ($j('#posttext').length > 0) {
				var type = 'before';
				var node = $j('#posttext');
			} else if ($j('#bbp_topic_content').length > 0) {
				var type = 'before';
				var node = $j('#bbp_topic_content');
			} else if ($j('#bbp_reply_content').length > 0) {
				var type = 'before';
				var node = $j('#bbp_reply_content');
			}	else {
				return;
			}
			switch (type) {
				case 'after':
					node.after(grinsDiv);
					$j("#wp_grins").css("paddingTop", "5px");
					break;
				case 'before':
					node.before(grinsDiv);
					break;
			}
		} //end success
		$j.ajax(s);
	}
};
	$j.wpgrins.init();
});

wp-grins.php
<?php
/*
Plugin Name: SSL Grins
Plugin URI: http://halfelf.org/plugins/wp-grins-ssl
Description: A Clickable Smilies hack for WordPress.
Version: 5.3.2
Author: Alex King, Ronald Huereca, Mika Epstein
Author URI: http://www.ipstenu.org
Text Domain: wp-grins-ssl

	Original plugin WP Grins Copyright (c) 2004-2007 Alex King
	http://alexking.org/projects/wordpress
	RIP Alex

	SSL version created on June 20, 2008 by Ronald Huereca
	SSL fork created on Sept 21, 2011 by Mika "Ipstenu" Epstein
	Copyright 2011-2016 Mika Epstein (email: ipstenu@ipstenu.org)

    This file is part of SSL Grins, a plugin for WordPress.

    SSL Grins is free software: you can redistribute it and/or
	modify it under the terms of the GNU General Public License as published
	by the Free Software Foundation, either version 2 of the License, or
    (at your option) any later version.

    SSL Grins is distributed in the hope that it will be
    useful, but WITHOUT ANY WARRANTY; without even the implied warranty
    of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with WordPress.  If not, see <http://www.gnu.org/licenses/>.

*/

global $wp_version;

if (version_compare($wp_version,"3.8","<")) { 
	exit( __('This plugin requires WordPress 3.8', 'wp-grins-ssl') ); 
}

if (!class_exists('WPGrinsHELF')) {
    class WPGrinsHELF {

        static $wpg_ver = '5.3.2'; // Plugin version so I can be lazy
		var $wpgs_defaults;
		var $wpgs_bbp_fancy;

        public function __construct() {
            add_action( 'init', array( $this, 'init' ) );

    		$this->wpgs_defaults = array(
    	        'comments'      => '1',
    	        'bbpress'       => '0',
    	        'buddypress'    => '0',
    	    );
    	    $this->bcq_bbp_fancy = get_option('_bbp_use_wp_editor');

        }

        public function init() {
			add_action( 'admin_init', array( $this,'admin_init' ) );
            add_filter( 'plugin_row_meta', array( $this, 'donate_link'), 10, 2);
            add_filter( 'plugin_action_links', array( $this, 'add_settings_link'), 10, 2 );

			if( !is_admin() && !in_array( $GLOBALS['pagenow'], array( 'wp-login.php', 'wp-register.php' )) ) {
    			add_action('wp_print_scripts', array( $this,'add_scripts_frontend'),1000);
    			add_action('wp_print_styles', array( $this,'add_styles_frontend'));
			}
			add_action('wp_ajax_grins', array( $this,'ajax_print_grins'));
			add_action('wp_ajax_nopriv_grins', array( $this,'ajax_print_grins'));
		}

		function ajax_print_grins() {
			echo $this->wp_grins();
			exit;
		}

		function wp_grins() {
    		global $wpsmiliestrans, $wp_version;

			$grins = '';
			$smiled = array();
			$smiled_hidden = array("bear", "wordpress", "martini", "developer", "whiterussian", "burrito", "facepalm", "kitten", "uneasy");

			foreach ($wpsmiliestrans as $tag => $grin) {

				if ( in_array($grin, $smiled_hidden) ) {
					$display = 'wpgs-hide';
				} else {
					$display = 'wpgs-default';
				}

                if (!in_array($grin, $smiled) ) {
  					$smiled[] = $grin;
   					$tag = esc_html(str_replace(' ', '', $tag));
   					$srcurl = apply_filters('smilies_src', includes_url("images/smilies/$grin"), $grin, site_url());

			        if ( is_plugin_active('new-smileys/new-smileys.php') || is_plugin_active('new-smileys-master/new-smileys.php') ) {
    						$grins .= "<span class='wp-smiley emoji emoji-$grin $display' alt='$tag' title='$grin' onclick='jQuery.wpgrins.grin(\"$tag\");'>$tag</span>";
                    } elseif ( $wp_version >= 4.2 ) {
						$img = $grin;
						$matches = array();
						$ext = preg_match( '/\.([^.]+)$/', $img, $matches ) ? strtolower( $matches[1] ) : false;
						$image_exts = array( 'jpg', 'jpeg', 'jpe', 'gif', 'png' );

						if ( ! in_array( $ext, $image_exts ) ) {
							$grins .= "<span class='wp-smiley twemoji' onclick='jQuery.wpgrins.grin(\"$tag\");' >$grin</span>";
						} else {
							$grins .= "<img src='$srcurl' alt='$tag' title='$grin' onclick='jQuery.wpgrins.grin(\"$tag\");' />";
						}
                    } else {
    					$grins .= "<img src='$srcurl' alt='$tag' title='$grin' onclick='jQuery.wpgrins.grin(\"$tag\");' />";
                    }
                }
			}
			return $grins;
		}

		function add_styles() {
			wp_enqueue_style('wp-grins', plugins_url( 'wp-grins.css', __FILE__ , '', self::$wpg_ver ) );
		}
		function add_scripts() {
		    wp_enqueue_script('wp_grins_ssl', plugins_url( 'wp-grins.js', __FILE__), array( 'jquery' ), self::$wpg_ver );
			wp_localize_script( 'wp_grins_ssl', 'wpgrinsssl', $this->get_js_vars());
		}

		function add_styles_frontend() {
    		$options = wp_parse_args(get_option( 'ippy_wpgs_options'), $this->wpgs_defaults );

    		if ( function_exists('is_bbpress') ) {
                if ( is_bbpress()  && ( $options['bbpress'] != '0') && !is_null($options['bbpress']) && ( $this->bcq_bbp_fancy == false ) ) {
                    $this->add_styles();
                }
              }
            if ( comments_open() && is_singular() && ( $options['comments'] != '0') && !is_null($options['comments']) ) {
                $this->add_styles();
            }
        }
		function add_scripts_frontend() {
    		$options = wp_parse_args(get_option( 'ippy_wpgs_options'), $this->wpgs_defaults );

    		if ( function_exists('is_bbpress') ) {
                if ( is_bbpress()  && ( $options['bbpress'] != '0') && !is_null($options['bbpress']) && ( $this->bcq_bbp_fancy == false ) ) {
                    $this->add_scripts();
                }
              }
            if ( comments_open() && is_singular() && ( $options['comments'] != '0') && !is_null($options['comments']) ) {
                $this->add_scripts();
            }
        }

        function get_js_vars() {
            if (is_ssl()) { $schema_ssl = 'https'; }
            else { $schema_ssl = 'http'; }
            return array(
                'Ajax_Url' => admin_url('admin-ajax.php', $schema_ssl),
                'LOCATION' => 'admin'
            );
        }

    	function admin_init(){
    		register_setting(
    			'discussion',               // settings page
    			'ippy_wpgs_options',         // option name
    			array( $this, 'validate_options') // validation callback
    		);

    		add_settings_field(
    			'ippy_wpgs_bbpress',         // id
    			__('SSL Grins', 'wp-grins-ssl'),                // setting title
    			array( $this, 'setting_input' ),   // display callback
    			'discussion',               // settings page
    			'default'                   // settings section
    		);
    	}

	    	// Display and fill the form field
	    	function setting_input() {
	    		$options = wp_parse_args(get_option( 'ippy_wpgs_options'), $this->wpgs_defaults );
	    		?>
	    		<a name="wpgs" value="wpgs"></a><input id='comments' name='ippy_wpgs_options[comments]' type='checkbox' value='1' <?php checked( $options['comments'], 1 ); ?> /> <?php _e('Activate Smilies for comments', 'wp-grins-ssl'); ?>
	    		<?php
	    		if ( function_exists('is_bbpress') && ( $this->bcq_bbp_fancy == false ) ) { ?>
	    		  <br /><input id='bbpress' name='ippy_wpgs_options[bbpress]' type='checkbox' value='1' <?php checked( $options['bbpress'], 1 ); ?> /> <?php _e('Activate Smilies for bbPress', 'wp-grins-ssl'); }
	    	}

	    	// Validate user input
	    	function validate_options( $input ) {
	    	    $options = wp_parse_args(get_option( 'ippy_wpgs_options'), $this->wpgs_defaults );
	    		$valid = array();
	
	    	    foreach ($options as $key=>$value) {
	        	    if (!isset($input[$key])) $input[$key]=$this->wpgs_defaults[$key];
	            }
	
	    	    foreach ($options as $key=>$value) {
	        	    $valid[$key] = $input[$key];
	            }
	
	    		unset( $input );
	    		return $valid;
	    	}

		// donate link on manage plugin page
		function donate_link($links, $file) {
    	        if ($file == plugin_basename(__FILE__)) {
    	                $donate_link = '<a href="https://store.halfelf.org/donate/">' . __( 'Donate', 'wp-grins-ssl' ) . '</a>';
    	                $links[] = $donate_link;
    	        }
    	        return $links;
    		}

	    	// add settings to manage plugin page
	    	function add_settings_link( $links, $file ) {
	    		if ( plugin_basename( __FILE__ ) == $file ) {
	    			$settings_link = '<a href="' . admin_url( 'options-discussion.php' ) . '#wpgs">' . __( 'Settings', 'wp-grins-ssl' ) . '</a>';
	    			array_unshift( $links, $settings_link );
	    		}
	    		return $links;
	    	}
    }
}

//instantiate the class
if (class_exists('WPGrinsHELF')) {
	new WPGrinsHELF();
}